packages/core/src/node/backend-application.ts

```diff
@@ -64,6 +70,14 @@ export class BackendApplication {
             }
         });
 
+        // Handles normal process termination.
+        process.on('exit', () => this.onStop());
+        // Handles `Ctrl+C`.
+        process.on('SIGINT', () => this.onStop());
+        // Handles `kill pid`
+        process.on('SIGUSR1', () => this.onStop());
```

Comment:
kill pid sends SIGTERM, not SIGUSR...

Reply:
Are you sure? I am not, after all. See: https://stackoverflow.com/questions/14031763/doing-a-cleanup-action-just-before-node-js-exits

Reply:
Yes I'm sure, this answer is wrong. See https://linux.die.net/man/1/kill
`The  default signal for kill is TERM`

Reply:
SIGURS are user-defined:

```
       SIGUSR1   30,10,16    Term    User-defined signal 1
       SIGUSR2   31,12,17    Term    User-defined signal 2

```

So we should not exit on those see: http://man7.org/linux/man-pages/man7/signal.7.html